internal/provider/azure/environ.go

```diff
@@ -233,19 +233,64 @@ func (env *azureEnviron) initEnviron(ctx context.Context) error {
 // PrepareForBootstrap is part of the Environ interface.
 func (env *azureEnviron) PrepareForBootstrap(ctx environs.BootstrapContext, _ string) error {
 	if ctx.ShouldVerifyCredentials() {
-		if err := verifyCredentials(env, ctx); err != nil {
+		if err := verifyCredentials(ctx, env); err != nil {
 			return errors.Trace(err)
 		}
 	}
 	return nil
 }
 
-// Create is part of the Environ interface.
-func (env *azureEnviron) Create(ctx envcontext.ProviderCallContext, args environs.CreateParams) error {
-	if err := verifyCredentials(env, ctx); err != nil {
+// ValidateModelCreation is part of the [environs.ModelResources] interface.
+func (env *azureEnviron) ValidateModelCreation(ctx context.Context) error {
+	if env.config.resourceGroupName == "" {
+		return nil
+	}
+	return env.validateResourceGroup(ctx, env.resourceGroup)
+}
+
+func (env *azureEnviron) validateResourceGroup(ctx context.Context, resourceGroupName string) error {
+	resourceGroups, err := env.resourceGroupsClient()
+	if err != nil {
 		return errors.Trace(err)
 	}
-	return errors.Trace(env.initResourceGroup(ctx, args.ControllerUUID, env.config.resourceGroupName != "", false))
+	logger.Debugf(ctx, "validating existing resource group %q for model %q", resourceGroupName, env.modelName)
+	g, err := resourceGroups.Get(ctx, resourceGroupName, nil)
+	if err != nil {
+		return env.HandleCredentialError(ctx, errors.Annotatef(err, "checking resource group %q", resourceGroupName))
+	}
+	if region := toValue(g.Location); region != env.location {
+		return errors.Errorf("cannot use resource group in region %q when operating in region %q", region, env.location)
+	}
+	modelTag := names.NewModelTag(env.config.Config.UUID())
+	if err := env.checkResourceGroup(g.ResourceGroup, modelTag); err != nil {
+		return env.HandleCredentialError(ctx, errors.Annotate(err, "validating resource group"))
+	}
+	return nil
+}
+
+// CreateModelResources is part of the [environs.ModelResources] interface.
+func (env *azureEnviron) CreateModelResources(ctx context.Context, args environs.CreateParams) error {
+	if err := verifyCredentials(ctx, env); err != nil {
```

Comment:
Something to think about.

I don't agree with this check. I know it is pre existing but all it does is validate that at this exact point in time things work. Because you have no way of making this an atomic operation it is pointless as the rest of the code is designed to report credential problems upwards.

Reply:
I have removed all the others, meant to remove this one too, but got lost in all the refactoring on the azure side